sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -200,214 +182,234 @@ export class EventHubReceiver extends LinkEntity {
 
-      if (receiver && !receiver.isSessionItselfClosed()) {
-        if (!this.isConnecting) {
-          log.error(
-            "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-              "address '%s' and the sdk did not initiate this. Hence calling detached from the " +
-              "_onSessionClose() handler.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-          await this.onDetached(sessionError);
-        } else {
-          log.error(
-            "[%s] 'session_close' event occurred on the session of receiver '%s' with " +
-              "address '%s' and the sdk did not initiate this. Moreover the receiver is already " +
-              "re-connecting. Hence not calling detached from the _onSessionClose() handler.",
-            this._context.connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else {
-        if (this._abortSignal) {
-          this._abortSignal.removeEventListener("abort", this._onAbort);
-        }
-        log.error(
-          "[%s] 'session_close' event occurred on the session of receiver '%s' with address " +
-            "'%s' because the sdk initiated it. Hence not calling detached from the _onSessionClose" +
-            "() handler.",
-          this._context.connectionId,
-          this.name,
-          this.address
-        );
-      }
-    };
+    const amqpError = amqpReceiver.error;
+    if (!amqpError) {
+      return;
+    }
+
+    const error = translate(amqpError);
+    log.error(
+      "[%s] An error occurred for Receiver '%s': %O.",
+      this._context.connectionId,
+      this.name,
+      error
+    );
+
+    if (!amqpReceiver.isItselfClosed() && this._onError) {
+      log.error(
+        "[%s] Since the user did not close the receiver " +
+          "we let the user know about it by calling the user's error handler.",
+        this._context.connectionId
+      );
+      this._onError(error);
+    } else {
+      log.error(
+        "[%s] The receiver was closed by the user." +
+          "Hence not notifying the user's error handler.",
+        this._context.connectionId
+      );
+    }
+  }
+
+  private _onAmqpSessionError(context: EventContext): void {
+    const amqpReceiver = this._receiver || context.receiver;
+    if (!amqpReceiver) {
+      return;
+    }
+
+    const sessionError = context.session && context.session.error;
+    if (!sessionError) {
+      return;
+    }
+
+    const error = translate(sessionError);
+    log.error(
+      "[%s] An error occurred on the session for Receiver '%s': %O.",
+      this._context.connectionId,
+      this.name,
+      error
+    );
+    if (!amqpReceiver.isSessionItselfClosed() && this._onError) {
+      log.error(
+        "[%s] Since the user did not close the receiver and the session error is not " +
+          "retryable, we let the user know about it by calling the user's error handler.",
```

Comment:
The retryable-ness of the error is checked elsewhere. We should update the logs here and add logs in the other place

Reply:
I see the logs removed from here, but it isnt added to the other place ...